projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.ts

```diff
@@ -22,86 +31,102 @@ export interface PropertyInfoValues {
-
-  ngOnChanges() {
-    this.getResource(this.iri);
-  }
-
-  /**
-   * Get a read resource sequence with ontology information and incoming resources.
-   *
-   * @param iri resourceIri
-   */
-  getResource(iri: string): void {
-
-    this.knoraApiConnection.v2.res.getResource(iri).subscribe(
-      (response: ReadResource) => {
-        this.resource = response;
-
-        // gather resource property information
-        this.resPropInfoVals = this.resource.entityInfo.classes[this.resource.type].getResourcePropertiesList().map(
-            (prop: IHasPropertyWithPropertyDefinition) => {
-                const propInfoAndValues: PropertyInfoValues = {
-                    propDef: prop.propertyDefinition,
-                    guiDef: prop,
-                    values: this.resource.getValues(prop.propertyIndex)
-                };
-                return propInfoAndValues;
-            }
-        );
-
-        // sort properties by guiOrder
-        this.resPropInfoVals.sort((a, b) => (a.guiDef.guiOrder > b.guiDef.guiOrder) ? 1 : -1);
-
-        // get system property information
-        this.systemPropDefs = this.resource.entityInfo.getPropertyDefinitionsByType(SystemPropertyDefinition);
-
-        // set the arkUrl value
-        this.versionArkUrl = this.resource.versionArkUrl;
-
-      },
-      (error: ApiResponseError) => {
-        console.error('Error to get resource: ', error);
-      });
-  }
-
-  /**
-   * Display message to confirm the copy of the citation link (ARK URL)
-   * @param message
-   * @param action
-   */
-  openSnackBar(message: string, action: string) {
-    message = 'Copied to clipboard!';
-    action = 'Citation Link';
-    this._snackBar.open(message, action, {
-      duration: 3000,
-      horizontalPosition: 'center',
-      verticalPosition: 'top'
-    });
-  }
+export class ResourceViewComponent implements OnInit, OnChanges, OnDestroy {
+
+    /**
+     * Resource iri
+     *
+     * @param [iri] Resource iri
+     */
+    @Input() iri: string;
+
+    resource: ReadResource;
+
+    resPropInfoVals: PropertyInfoValues[] = []; // array of resource properties
+
+    systemPropDefs: SystemPropertyDefinition[] = []; // array of system properties
+
+    versionArkUrl: string; // versionArkUrl value
+    message: string; // message to show in the snackbar to confirm the copy of the ARK URL
+    action: string; // label for the snackbar action
+
+    eventBusSubscription: Subscription;
+
+    constructor(
+        @Inject(DspApiConnectionToken) private knoraApiConnection: KnoraApiConnection,
+        private _snackBar: MatSnackBar,
+        private _eventBusService: EventBusService) { }
+
+    ngOnInit() {
+        // subscribe to the event bus and listen for the ValueAdded event to be emitted
+        // when a ValueAdded event is emitted, get the resource again to display the newly created value
+        // TODO: find a better way to show the new value than having to get the entire resource again
```

Comment:
Why does the resource have to be loaded again? The newly created value is already read back from Knora, right?

Reply:
it's true that the newly created value is read back from knora, but I can't find a way to updated the UI to show the new value besides reloading the entire resource

Reply:
Ok, please make a DSP task so we can figure out a better way sometime.

Reply:
it's in the list of tasks for V2 https://dasch.myjetbrains.com/youtrack/issue/DSP-446

Reply:
ok, great.